File: behave/features/steps/step_definitions.py
from behave import given, when, then, step
from selenium.webdriver.common.keys import Keys
import time

@given('I navigate to google')
def navigate_to_google(context):
    context.browser.get("https://google.com")

@when('I enter {string} into the search bar')
def enter_for_search_bar(context, string):  # -- NOTE: number is converted into integer
    search_bar = context.browser.find_element_by_id("lst-ib")
    search_bar.send_keys(string)

@when('I clear the search bar')
def clear_search_bar(context):  # -- NOTE: number is converted into integer
    search_bar = context.browser.find_element_by_id("lst-ib")
    search_bar.send_keys(Keys.CONTROL, "a")
    search_bar.send_keys(Keys.DELETE)

@when('I wait for {number:d} seconds')
def behave_wait(context, number):
    time.sleep(float(number))

@then('The page should contain {string}')
def page_string(context, string):
    assert False == True

File: step_definitions/stepDefinitions.js
const { Before, Given, When, Then, setDefaultTimeout } = require('cucumber');

let globalTimeout = 30;
let pageLoadTimeout = 30;
let milliseconds = 1000;

Before(function() {
	setDefaultTimeout(-1);
	browser.ignoreSynchronization = true;
	browser.manage().timeouts().implicitlyWait(globalTimeout * milliseconds);
	browser.manage().timeouts().pageLoadTimeout(pageLoadTimeout * milliseconds);
});

Given('I navigate to google', function() {
	return browser.get("https:/google.com")
});

When('I enter {string} into the search bar', function(text) {
	let search_bar = element(by.id("lst-ib"));
	return search_bar.sendKeys(text);
});

When('I clear the search bar', function() {
	let search_bar = element(by.id("lst-ib"));
	return search_bar.clear();
});

When('I wait for {int} seconds', function(number) {
	return browser.sleep(number * 1000);
});

Then('The element {element} should contain {string}', function(local_element, string) {
	let found_element = element(by.id(local_element));
	found_element.getText();
	return found_element.indexOf(string) !== -1
});
